### PR TITLE
Avoid a specific case where `conda env update` hangs.

### PR DESCRIPTION
**Pull request by fragmede**
## Description
When using an environment.yml that asks pip to clone a git repo and that path exists, pip will ask for user input. As we are not in a context to provide such, tell pip what to do so it doesn't hang on waiting for user input.

### Setup:

environment.yml has
```
name: exists
dependencies:
  - python
  - pip
  - pip:
    - -e git+https://github.com/conda/conda
 ```

1. Run `conda env create`.
1. Run `conda env update`.

The env create runs fine. The conda env update, however, will hang while displaying a spinner . Hacking the code to [say what it's doing](https://github.com/fragmede/conda/commit/e4862e20508b75144e067be0aba3ce0e122565e7) (might want to add that as a debug option) show that the pip process is asking for user input:  `What to do?  (s)witch, (i)gnore, (w)ipe, (b)ackup`, which corresponds to the files existing. See the [pip documentation](https://pip.pypa.io/en/stable/cli/pip/#exists-action-option).

This patch force specifies that we want to backup the existing files and prevents pip from asking for user input in this specific case.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Review by kenodegard — CHANGES_REQUESTED** [review]
@fragmede a reasonable addition, please include a news snippet and this should be good to go

**Comment by fragmede** [comment]
@kenodegard I have added news file news/11818-conda-hang.

**Review by jezdez — APPROVED** [review]
This looks good, thank you!

File: conda_env/installers/pip.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def _pip_install_via_requirements(prefix, specs, args, *_, **kwargs):
         requirements.write('\n'.join(specs))
         requirements.close()
         # pip command line...
-        pip_cmd = ['install', '-U', '-r', requirements.name]
+        pip_cmd = ['install', '-U', '-r', requirements.name, '--exists-action=b']
```
Comment (kenodegard) [inline conda_env/installers/pip.py:51]:
```suggestion
        # see https://pip.pypa.io/en/stable/cli/pip/#exists-action-option
        pip_cmd = ['install', '-U', '-r', requirements.name, '--exists-action=b']
```

File: news/11818-conda-hang [hunk 2]
```diff
@@ -0,0 +1,21 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* conda env update would ask for user input and hang when working with pip
+installs of git repos and the repo was previously checked out. Tell pip not to
+ask for user input for that case.  (#11818)
```
Comment (kenodegard) [inline news/11818-conda-hang:9]:
```suggestion
* `conda env update` would ask for user input and hang when working with pip
installs of git repos and the repo was previously checked out. Tell pip not to
ask for user input for that case.  (#11818)
```

**Review by kenodegard — CHANGES_REQUESTED** [review]
Minor documentation additions

**Review by kenodegard — APPROVED** [review]
@fragmede thanks for working on this, it's really appreciated 🚀

**Pull request merged**